### ISSUE TITLE
Disable sorting of NodeSet?

### ISSUE THREAD

**Issue by brianjmurrell** [cea-hpc/clustershell#437]
I'd like to be able to disable `NodeSet` sorting so that, for example, `cluset -e -I 0 5,4,3,2,1` returns me `5`.

Is this possible?  If not, could it be possible?

**Comment by degremont** [comment]
This is not possible. I think this is a duplicate of #410. They both relies on the same internal behaviour of NodeSet which sorts, by design, the elements. 
It should be easier to implement that using an ordered dict or something similar in Python 3+.

**Comment by brianjmurrell** [comment]
So, yeah.  I thought I had bug reported that before, but I was using _sort_ as my search term, not _order_, which is why I didn't find it.

> It should be easier to implement that using an ordered dict or something similar in Python 3+.

Are you suggesting that is something that could be done under the hood here, or that it's something that should be done outside of clustershell?

If the latter (a) I am using the `cluset` command, not a Python script and (b) I still want all of the other nodeset handling (folding, expanding, etc.) that `cluset` provides.

Am I misunderstanding your suggestion?

**Comment by degremont** [comment]
It was more a note for anybody wanted to dig into the code side of this feature :)

Actually, the internal data structure of nodeset is a set, which is non-ordered, it is not possible to restore the original order in which the element were added. That's why we need to change this to a *ordered set* (or kind of), in the future to enable the feature you are looking for.

**Comment by brianjmurrell** [comment]
Fair enough.  Thanks for your time on this.  I will try to remember this issue in the future.

**Comment by thiell** [comment]
Closing as a duplicate of #410 — same underlying by-design NodeSet sorting behavior, discussed above.
